netadapter/server/grpcserver/grpc_server.go

```diff
@@ -0,0 +1,130 @@
+package grpcserver
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"google.golang.org/grpc/peer"
+
+	"github.com/kaspanet/kaspad/netadapter/server"
+	"github.com/kaspanet/kaspad/netadapter/server/grpcserver/protowire"
+	"github.com/kaspanet/kaspad/util/panics"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+)
+
+type gRPCServer struct {
+	onConnectedHandler server.OnConnectedHandler
+	connections        map[string]*gRPCConnection
+	listeningAddrs     []string
+	server             *grpc.Server
+}
+
+// NewGRPCServer creates and starts a gRPC server with the given
+// listening port
```

Comment:
Please correct this comment. There are now listeningAddrs instead of a single listening port.